cts/cts-pengine-autofix.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/python
```

Comment:
I'm going to merge this to proceed with rc2, but I'll remove this file before release, so I can examine it more closely later. I don't want to make such changes to regression tests easy -- someone needs to have a firm understanding of the issues before mucking with test input.

General comments:

I prefer that executables (as opposed to importable modules) not have a language extension such as ".sh" or ".py". The language is an implementation detail that can theoretically change.

We now keep all Python executables as .in files and use #!%PYTHON% for the shebang, to allow configurability of python version and location. They must have an AC_CONFIG_FILES() line in configure.ac, and the file name without .in in .gitignore.

For something as potentially dangerous as changing regression tests, dry run should be the default, and an option should be required to make it live.

Reply:
I see some explanations are owed; I don't think removing that file is the best
thing to do, really:

* it's not meant to make the jumps *performed lightly*, but *performed easily*
  once the decision is made and there's no other way around (but to perform
  the same suggested steps manually); it helped me significantly to get rid of
  the mental strain that I will copy/paste something wrongly or will introduce
  hard-to-notice (considering the excessively long lines) unintended changes,
  plus it allowed me to return and tight the digest changes right at the suitable
  commit (yep,  time investment likely did not pay off for this first time of use,
  but it eventually will, I am sure)

* it's not meant to be distributed nor packaged, it's only to be used by expert
  person and directly from the repo checkout
  - this will hopefully justify that "dry run" is not the default -- I only added it
    since the triviality of the feature was begging for it and that it might come
    handy under some circumstances, but there's 99% chance it won't be
    the case inside said repo checkout -- tuned `git diff`ing is far superior
    way to review the changes made compared to basic output of Python's
    standard `difflib`
  - this will also hopefully justify I didn't want to involve `configure` machinery
    here as opposed to ready-to-use script, and in fact, I originally had to
    hashbang nor the script was executable -- would it then be acceptable to
    revert such a state, assuming that extra extension + non-executability
    will indicate (to the audience of the repo only!) clearly this is something
    special not to be confused with a proper test suite runner?
    (Also it should work with any frequented version of Python today,
    and said expert person is expected to know how to apply it.)

* regarding theoretical implementation change, true, but plain shell
   would become cumbersome easily should we want to add a new
   regexp-heavy fixer (also consider problem with sed script portability),
   Perl is kept away now, and compiled program is overkill for something
   used so rarely and not distributed conventionally at all; again deferring
   to the expect person intended use case -- they will be able to
   accommodate should the language and hence the extension change